src/bin/cobalt/new.rs

```diff
@@ -409,8 +419,26 @@ pub fn publish_document(file: &path::Path) -> Result<()> {
     let doc =
         cobalt_model::DocumentBuilder::<cobalt_model::FrontmatterBuilder>::new(front, content);
     let doc = doc.to_string();
-
     cobalt_model::files::write_document_file(doc, file)?;
 
+    if config.posts.add_date_prefix {
+        // avoid prepend to existing date prefix
+        let (_, file_name) = cobalt_model::parse_file_stem(
+            file.file_stem()
+                .map(|os| os.to_string_lossy().into_owned())
+                .unwrap_or_else(|| "".to_owned()),
+        );
+
+        let file_name = format!(
+            "{}{}.{}",
+            date.prefix(),
+            file_name,
+            file.extension()
+                .map(|os| os.to_string_lossy().into_owned())
+                .unwrap_or_else(|| config.posts.template_extensions[0].to_owned())
```

Comment:
1. Is it possible to get in this situation? I don't remember what the CLI looks like for this.
2. We should have an error check somewhere to ensure `template_extensions` has at least one item in it

Reply:
I don't know if cobalt can accept files without extension, can it?

I didn't check `template_extensions` because default value put `md` and `liquid` in it.

Reply:
People can override defaults

Reply:
If no extension is defined, what shall we do? 
another format for the file name `{}{}` instead of `{}{}.{}`?

Reply:
Maybe we should error for now?

I don't thinks can work correctly without an extension, for example, I don't think a file can be included in a collection without one.  This will also be easier to change in the future and less surprising to users (than us adding one).

We add an extension in new and rename but I think that is a CLI usability thing.  In publish, they already have a file and we shouldn't change it behind their back.